docs/topics/forms/media.txt

```diff
@@ -341,3 +341,35 @@ form::
     <script type="text/javascript" src="http://static.example.com/animations.js"></script>
     <script type="text/javascript" src="http://static.example.com/actions.js"></script>
     <script type="text/javascript" src="http://static.example.com/whizbang.js"></script>
+
+Custom Media Types
+------------------
+
+By subclassing ``Media``, you can add support for media types beyond the
+default ``css`` and ``js`` types. One use case for this would be adding support
+for inline JavaScript.
+
+To add a custom media type, you need to add the type to the ``MEDIA_TYPES``
+tuple and add ``add_X`` and ``render_X`` methods for adding and rendering
+your custom type. For example::
+
+    from django.utils.html import format_html
+
+    class InlineJSMedia(Media):
+        # MEDIA_TYPES stores tuples of (media_name, constructor).
+        # __init__ will create an attribute self._media_name using the
+        # constructor.
+        MEDIA_TYPES = Media.MEDIA_TYPES + (('inline_js', list),)
```

Comment:
Needs documentation update for `media_types`: https://docs.djangoproject.com/en/dev/topics/forms/media/#assets-as-a-static-definition